packages/presentation/lib/home_screen/bloc/palindrome_bloc.dart

```diff
@@ -0,0 +1,33 @@
+import 'dart:async';
+
+import 'package:domain/use_cases/impl/palindrome_impl.dart';
+
+class PalindromeBlock {
+  final PalindromeCaseImpl _palindromeUseCase;
+
+  PalindromeBlock(this._palindromeUseCase) {
+    _inputStreamController.stream.listen(_checkPalindrome);
```

Comment:
why do u need the stream here to pass events from widget to bloc if u can call bloc methods directly?